app/controllers/welcome_controller.rb

```diff
@@ -1,64 +1,17 @@
 class WelcomeController < ApplicationController
-  skip_before_action :verify_authenticity_token, only: [:show]
+  skip_before_action :verify_authenticity_token
 
   def show
-    payload = ApplicationKey.validate_jwt!(params[:jwt]) if params[:jwt]
+    payload = get_payload(params[:jwt])
 
     redirect_to after_login_path(payload, current_user) and return if current_user
 
-    @email = params.dig(:user, :email)
-    if @email
-      if @email.blank?
-        @email_error_message = I18n.t("activerecord.errors.models.user.attributes.email.blank")
-        return
-      end
-
-      unless Devise.email_regexp.match? @email
-        @email_error_message = I18n.t("activerecord.errors.models.user.attributes.email.invalid")
-        return
-      end
-
-      if User.exists?(email: @email)
-        login_state = create_login_state(payload, @email)
-        session[:login_state_id] = login_state.id
-        redirect_to user_session_path
-      elsif Rails.configuration.enable_registration
-        if payload || !Rails.configuration.force_jwt_at_registration
-          registration_state = create_registration_state(payload, @email)
-          session[:registration_state_id] = registration_state.id
-          redirect_to new_user_registration_start_path
-        else
-          render "devise/registrations/transition_checker"
-        end
-      else
-        render "devise/registrations/closed"
-      end
-    end
+    redirect_to new_user_registration_start_path
```

Comment:
After #478 lands, this redirect will need to preserve the `_ga` query param

Reply:
I guess it'll also need adding to the ["Sign in" link](https://github.com/alphagov/govuk-account-manager-prototype/pull/476/files#diff-3e187d20d1ccdc19016724e4f655fbea393c5e1324138a2765fbebc94789444dR66) on the registration form too?